src/dftbp/dftbplus/parser.F90

```diff
@@ -6208,26 +6204,18 @@ subroutine reduceGeometry(contactVec, contactRange, lateralContactSeparation, ge
     if (geom%tPeriodic) then
       contUnitVec = contactVec / sqrt(sum(contactVec**2, dim=1))
       dots = abs(matmul(contUnitVec, geom%latVecs))
-      mask = (abs(dots - sqrt(sum(geom%latVecs, dim=1)**2)) < 1e-8_dp)
+      mask(:) = (abs(dots - sqrt(sum(geom%latVecs, dim=1)**2)) < 1e-8_dp)
       if (count(mask) /= 1) then
         call error("Too many lattice vectors parallel to the contact")
       end if
-      ! Workaround for bug in Intel compiler (can not use index function)
```

Comment:
Which Intel versions are actually affected?

Reply:
5+ years old, the original instance being in commit 3226cf2031 from 2017-10-26 and probably a mistake to call it a workaround, since that isn't what the index() function does. It's for strings, but might have been a compiler extension somewhere to work for logicals at some point.